areaOfInterest/areas-of-interest.ts

```diff
@@ -0,0 +1,103 @@
+class AreasOfInterest {
+
+    preInit(config: any) {
+        this.config = config;
+    }
+
+    init(api: any) {
+        console.log(api);
+        //initialize attributes
+        let map = api; // map object that rv-extension loads into this file
+        let mapConfig = this.config;
+        let zones = mapConfig.map.components.areaOfInterest._source.areas;
+        let noPic = mapConfig.map.components.areaOfInterest._source.noPicture;
+        let zoneList = ' ';
+        let title = this.translations[mapConfig.language];
+
+        //define and append setExtent to head
+        let setExtent = `
```

Comment:
Find a way to inline this code so a script tag does not need to be added to the header. Try `map.extent = ...`.

Reply:
done!